plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/AltibaseSQLDialect.java

```diff
@@ -0,0 +1,626 @@
+    }
+
+    @NotNull
+    private SQLTokenPredicateSet makeDialectSkipTokenPredicates(JDBCDataSource dataSource) {
+        SQLSyntaxManager syntaxManager = new SQLSyntaxManager();
+        syntaxManager.init(this, dataSource.getContainer().getPreferenceStore());
+        SQLRuleManager ruleManager = new SQLRuleManager(syntaxManager);
+        ruleManager.loadRules(dataSource, false);
+        TokenPredicateFactory tt = TokenPredicateFactory.makeDialectSpecificFactory(ruleManager);
+
+        TokenPredicateSet conditions = TokenPredicateSet.of(
+                /* Oracle grammar */
+                new TokenPredicatesCondition(
+                        SQLParserActionKind.BEGIN_BLOCK,
+                        tt.sequence(
+                                "CREATE",
+                                tt.optional("OR", "REPLACE"),
+                                "PACKAGE", "BODY"
+                        ),
+                        tt.sequence()
+                ),
+                new TokenPredicatesCondition(
+                        SQLParserActionKind.SKIP_SUFFIX_TERM,
+                        tt.sequence(
+                                "CREATE",
+                                tt.optional("OR", "REPLACE"),
+                                tt.alternative("FUNCTION", "PROCEDURE")
+                        ),
+                        tt.sequence(
+                                tt.alternative(
+                                        tt.sequence("RETURN", SQLTokenType.T_TYPE),
+                                        "deterministor", "pipelined", "parallel_enable", "result_cache",
+                                        ")",
+                                        tt.sequence("procedure", SQLTokenType.T_OTHER),
+                                        tt.sequence(SQLTokenType.T_OTHER, SQLTokenType.T_TYPE)
+                                        ), ";")
+                ),
+                new TokenPredicatesCondition(
+                    SQLParserActionKind.BEGIN_BLOCK,
+                    tt.sequence(),
+                    tt.sequence(tt.not("END"), "IF", tt.not("EXISTS"))
+                ),
+                /*
+                 * Altibase only grammar
+                 * Grammar conflict between PSM and Typeset
+                 * PSM: CREATE...[AS|IS] ... BEGIN ... END
+                 * TYPESET: CREATE...[AS|IS] ... END
+                 */
+                new TokenPredicatesCondition(
+                        SQLParserActionKind.BEGIN_BLOCK,
+                        tt.sequence(
+                                "CREATE",
+                                tt.optional("OR", "REPLACE"),
+                                "TYPESET"
+                        ),
+                        tt.sequence()
+                ),
+                new TokenPredicatesCondition(
+                        SQLParserActionKind.SKIP_SUFFIX_TERM,
+                        tt.token("END"),
+                        tt.sequence(";", "/")
```

Comment:
I believe it's a script delimiters and only one of them should go at the end of query. Then you need to put them to alternative
```suggestion
                        tt.sequence(tt.alternative(";", "/"))
```

Reply:
After working on this plugin on and off for a long time, I forgot about this part. I will delete "/" as it is not an officially supported grammar.

To explain in detail, "/" is a special token used to signal the end of Procedure/Function/Typeset in a command line query tool called Altibase iSQL, and is not supported in official Altibase SQL grammar. The purpose of initially putting it here was to enable the existing iSQL script could be reused. DBeaver sends the input as it is, so a grammar error occurs if there is a "/" at the end. I'll think it over again.

Thanks again your kind review :)